pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -156,19 +176,30 @@ private StreamObserver<MailboxContent> getContentObserver() {
         .open(_statusObserver);
   }
 
-  private MailboxContent toMailboxContent(MseBlock block, List<DataBuffer> serializedStats)
+  private void splitAndSend(MseBlock block, List<DataBuffer> serializedStats)
       throws IOException {
     _statMap.merge(MailboxSendOperator.StatKey.RAW_MESSAGES, 1);
     long start = System.currentTimeMillis();
     try {
       DataBlock dataBlock = MseBlockSerializer.toDataBlock(block, serializedStats);
-      ByteString byteString = DataBlockUtils.toByteString(dataBlock);
-      int sizeInBytes = byteString.size();
+      List<ByteString> byteStrings;
+      if (_splitBlocks) {
```

Comment:
Consider moving this check one level up (into `sendInternal()`) to separate the handling of split vs non-split to reduce overhead of non-split case